daft/dataframe/dataframe.py

```diff
@@ -1011,26 +1011,16 @@ def collect(self, num_preview_rows: Optional[int] = 8) -> "DataFrame":
 
         return self
 
-    @DataframePublicAPI
-    def show(self, n: int = 8) -> "DataFrameDisplay":
-        """Executes enough of the DataFrame in order to display the first ``n`` rows
-
-        .. NOTE::
-            This call is **blocking** and will execute the DataFrame when called
-
-        Args:
-            n: number of rows to show. Defaults to 8.
-
-        Returns:
-            DataFrameDisplay: object that has a rich tabular display
-        """
+    def _construct_show_display(self, n: int) -> "DataFrameDisplay":
+        """Helper for .show() which will construct the underlying DataFrameDisplay object"""
         preview_partition = self._preview.preview_partition
         total_rows = self._preview.dataframe_num_rows
 
         # Truncate n to the length of the DataFrame, if we have it.
         if total_rows is not None and n > total_rows:
             n = total_rows
 
+        # Construct the PreviewPartition
         if preview_partition is None or len(preview_partition) < n:
```

Comment:
what happens here if the final result is less than `n`? for example
```
df = daft.read_parquet(...).count()
```
maybe we should have a another bit to track if this the preview partition is the full result

Reply:
See separate discussion where this logic came from: https://github.com/Eventual-Inc/Daft/pull/1651#discussion_r1401056465

Reply:
kk